docs/preprocessor/rewrite-links.py

```
@@ -0,0 +1,318 @@
+
+
+def split_link_target(target: str) -> tuple[str, str]:
+    for separator in ("#", "?"):
+        if separator in target:
+            index = target.find(separator)
+            return target[:index], target[index:]
+    return target, ""
+
+
+def normalize_repo_relative_path(
+    chapter_path: str, link_path: str, docs_root: Path, src_dir: str
```

Ruff (0.15.13)

[warning] Unused function argument: `src_dir`

(ARG001)